mindmeld/auto_annotator.py

```diff
@@ -835,5 +895,282 @@ def valid_entity_check(self, entity):
+        Returns:
+            selected_spans (list): List of the largest non-overlapping spans.
+        """
+        spans.sort(reverse=True)
+        selected_spans = []
+        for span in spans:
+            has_overlaps = [
+                span.has_overlap(selected_span) for selected_span in selected_spans
+            ]
+            if not any(has_overlaps):
+                selected_spans.append(span)
+        return selected_spans
+
+    @staticmethod
+    def _filter_duckling_candidates(candidates):
+        """Pipeline function to filter initial list of duckling candidates using heuristics.
+
+        Args:
+            candidates (list): List of duckling candidates
+        Returns:
+            filtered_candidates (list): List of filtered duckling candidates.
+        """
+        filtered_candidates = NoTranslationDucklingAnnotator._remove_unresolved_sys_amount_of_money(
+            candidates
+        )
+        return filtered_candidates
+
+    @staticmethod
+    def _remove_unresolved_sys_amount_of_money(candidates):
+        """Do not label candidate entities that are sys_amount-of-money but
+        do not have an "unknown" unit type.
+        """
+        return [
+            candidate
+            for candidate in candidates
+            if not (
+                candidate["dim"] == "amount-of-money"
+                and candidate["value"]["unit"] == "unknown"
+            )
+        ]
+
+
+class TranslationDucklingAnnotator(Annotator):
+    """Custom Annotator class used to generate annotations."""
+
+    def __init__(self, app_path, config=None, en_annotator=None):
+        """Initializes an annotator.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+        self.translator = TranslatorFactory().get_translator(
+            self.config.get("translator")
+        )
+        self.en_annotator = en_annotator or SpacyAnnotator(
+            self.app_path, self.config, language=ENGLISH_LANGUAGE_CODE
+        )
+
+    def parse(self, sentence, entity_types=None, language=None, locale=None, **kwargs):
+        """
+        Args:
+            sentence (str): Sentence to detect entities.
+            entity_types (list): List of entity types to parse. If None, all
+                    possible entity types will be parsed.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+        Returns: entities (list): List of entity dictionaries.
+        """
+        locale = locale or self.locale
+        language = language or self.language
```

Comment:
We should assert here that the language is not english

Reply:
Added